be/src/exprs/condition_expr.cpp

```diff
@@ -186,8 +192,12 @@ class VectorizedNullIfExpr : public Expr {
         return result.build(ColumnHelper::is_all_const(columns));
     }
 
-    ColumnPtr _evaluate_complex(const Columns& columns) {
-        auto num_rows = columns[0]->size();
+    ColumnPtr _evaluate_complex(const Columns& inputs) {
+        auto num_rows = inputs[0]->size();
+        Columns columns;
+        for (const auto& col : inputs) {
+            columns.push_back(ColumnHelper::unpack_and_duplicate_const_column(num_rows, col));
+        }
```

Comment:
there is no need to deal const_column for nullifexpr and coalesceexpr, they will use other branch.